internal/storage/db/storage.go

```diff
@@ -63,18 +64,27 @@ func (st *Storage) Ping(ctx context.Context) error {
 	return st.DB.PingContext(ctx)
 }
 
-func (st *Storage) Get(short string) (string, bool) {
-	var uri string
+func (st *Storage) Get(short string) (*storage.URIsItem, bool) {
+	var (
+		uri     string
+		deleted bool
+	)
 	err := st.DB.QueryRow(`
-		select uri
+		select uri, deleted
 		from uris t1
 		where t1.short = $1
-	`, short).Scan(&uri)
+	`, short).Scan(&uri, &deleted)
 	if err != nil {
 		log.Println(err)
-		return uri, false
-	}
-	return uri, true
+		return &storage.URIsItem{
+			URI:     uri,
+			Deleted: true,
+		}, false
```

Comment:
лучше возвращать ошибку выше и nil вместо значения.
```suggestion
		return nil, err
```
ведь по сути запись в базе не найдена, и так мы обезопасимся от работы со значением где-то выше